modules/op-resolver-bundle.adoc

```diff
@@ -0,0 +1,156 @@
+
+[id="resolver-bundles-config_{context}"]
+== Configuring the Bundles resolver
+
+Use the `TektonConfig` custom resource to configure the bundles resolver as necessary.
+
+To edit the `TektonConfig` custom resource, enter the following command:
+
+[source,bash]
+----
+oc edit TektonConfig config
+----
+
+In the `TektonConfig` custom resource, under `spec:` `pipeline:` `bundles-resolver-config:` specify configuration, as in the following example:
+
+[source,yaml]
+----
+apiVersion: operator.tekton.dev/v1alpha1
+kind: TektonConfig
+metadata:
+  name: config
+spec:
+  pipeline:
+    bundles-resolver-config:
+      default-service-account: pipelines
+----
+
+.Keys for configuring the Bundles resolver
+|===
+| Key | Description | Example value
+
+| `default-service-account`
+| The default service account name to use for bundle requests.
+| `default`, `someuser`
+
+| `default-kind`
+| The default layer kind in the bundle image.
+| `task`, `pipeline`
+|===
+
+[id="resolver-bundles-specify_{context}"]
+== Specifying a remote pipeline or task using the Bundles resolver
+
+To specify a remote pipeline or task from a Tekton bundle, use the following reference format under `pipelineRef` or `taskRef`:
+
+[source,yaml]
+----
+  resolver: bundles
+  params:
+  - name: bundle
+    value: <fully-qualified-name-of-image>
+  - name: name
+    value: <name-of-resource>
+  - name: kind
+    value: [pipeline|task]
+----
+
+.Supported parameters for the Bundles resolver
+|===
+| Parameter | Description | Example Value
+
+| `serviceAccount`
+| The name of the service account to use when constructing registry credentials
+| `default`
+
+| `bundle`
+| The bundle url pointing at the image to fetch
+| `gcr.io/tekton-releases/catalog/upstream/golang-build:0.1`
+
+| `name`
+| The name of the resource to pull out of the bundle
+| `golang-build`
+
+| `kind`
+| The kind of the resource to pull out of the bundle
+| `task`
+|===
+
+If the pipeline or task requires parameters, provide these parameters under `params`.
+
+.Example pipeline run that references a remote pipeline from a Tekton bundle
+[source,yaml]
+----
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: bundle-pipeline-reference-demo
+spec:
+  pipelineRef:
+    resolver: bundles
+    params:
+    - name: bundle
+      value: registry.example.com:5000/simple/pipeline:latest
+    - name: name
+      value: hello-pipeline
+    - name: kind
+      value: pipeline
+    - name: sample-pipeline-parameter
+      value: test
+  params:
+  - name: username
+    value: "pipelines"
+----
+
+.Example pipeline that references a remote task from a Tekton bundle
+[source,yaml]
+----
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: pipeline-with-bundle-task-reference-demo
+spec:
+  tasks:
+  - name: "bundle-task-demo"
+    taskRef:
+      resolver: bundles
+      params:
+      - name: bundle
+        value: registry.example.com:5000/simple/pipeline:latest
```

Comment:
We could use different reference for different example (just to show that they don't all need to be refering the same oci image).